proxy/healthy_endpoints.go

```diff
@@ -0,0 +1,35 @@
+package proxy
+
+import "github.com/zalando/skipper/routing"
+
+type healthyEndpoints struct {
+	enabled                         bool
+	timeoutedRequestsRatioThreshold float64
+	totalRequestsThreshold          float64
+}
+
+func (h *healthyEndpoints) healthyEndpoint(endpoint routing.LBEndpoint) bool {
+	timeoutedRequests := float64(endpoint.Metrics.TimeoutedRequests())
+	totalRequests := float64(endpoint.Metrics.TotalRequests())
+	failedRequestsRatio := timeoutedRequests / totalRequests
+
+	return failedRequestsRatio < h.timeoutedRequestsRatioThreshold || totalRequests < h.totalRequestsThreshold
+}
+
+func (h *healthyEndpoints) filterHealthyEndpoints(endpoints []routing.LBEndpoint, rt *routing.Route) []routing.LBEndpoint {
+	if !h.enabled {
+		return endpoints
+	}
+
+	filtered := make([]routing.LBEndpoint, 0, len(endpoints))
+	for _, e := range endpoints {
+		if h.healthyEndpoint(e) {
+			filtered = append(filtered, e)
+		}
```

Comment:
Once endpoint becomes unhealthy it stops getting requests and thus total and failed requests counters do not change. How will it recover?

Reply:
> it stops getting requests

I agree with your point, skipper still should send some minimum amount of requests even to unhealthy endpoints, see the latest `operation.md`.